app/src/main/java/com/bnyro/recorder/ui/MainActivity.kt

```diff
@@ -16,12 +16,13 @@ import com.bnyro.recorder.ui.models.ThemeModel
 import com.bnyro.recorder.ui.theme.RecordYouTheme
 
 class MainActivity : ComponentActivity() {
+    private var initialRecorder = RecorderType.NONE
+    private var exit = false
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
-
         val themeModel: ThemeModel by viewModels()
 
-        val initialRecorder = when (intent?.getStringExtra("action")) {
+        initialRecorder = when (intent?.getStringExtra("action")) {
```

Comment:
Please move "action" to a constant and re-use it everywhere else where we use "action".